lib/client/client.js

```diff
@@ -975,6 +988,9 @@ Client.prototype.connect = function connect () {
       maxDelay: this.reconnect.maxDelay
     })
     failAfter = this.reconnect.failAfter
+    if (this.urls.length > 1 && failAfter) {
+      failAfter *= this.urls.length
+    }
```

Comment:
if `reconnect` option is not specified `failAfter` will be equal to `1` so only the first url will be tried.

Reply:
Good point.. I'd say that while using multiple servers you always want to reconnect (meaning that disabling `reconnect` option does not make sense), but guess I shouldn't modify this behavior and just update `failAfter` to the number of servers in use.